### ISSUE TITLE
`RCOMBBASE` is broken

### ISSUE THREAD

**Issue by plafer** [0xPolygonMiden/miden-vm#1560]
Failing test:

```rs
#[test]
fn rcomb_base() {
    const X_PTR: u64 = 42;
    const Z_PTR: u64 = 142;
    const A_PTR: u64 = 1142;
    let asm_op = "begin rcomb_base end";

    let stack_inputs = [1,2,3,4,5,6,7,0,0,X_PTR, Z_PTR, A_PTR];
    let test = build_test!(asm_op, &stack_inputs);
    let _last_state = test.get_last_stack_state();
}
```

This test case also finds another bug in the memory chiplet [here](https://github.com/0xPolygonMiden/miden-vm/blob/766292d4efd9bcf663528631410a2d5361f3bb69/processor/src/chiplets/memory/mod.rs#L180-L182), where the fix is

```rs
let delta = if prev_ctx != ctx {
    (u32::from(ctx) - u32::from(prev_ctx)).into()
} else if prev_addr != addr {
    u64::from(addr - prev_addr)
} else if clk != prev_clk {
    clk - prev_clk - 1
} else {
    0
};
```

**Comment by bobbinth** [comment]
Interesting! I though we used it for Falcon signature verification - but I guess we do not.

**Comment by Al-Kindi-0** [comment]
Interestingly, we [do](https://github.com/0xPolygonMiden/miden-vm/blob/fbce798a53967310463ee02e0e681617781406df/stdlib/asm/crypto/dsa/rpo_falcon512.masm#L262)

**Comment by Al-Kindi-0** [comment]
From the error I am seeing, I think might be because the memory read is happening at 0-th cycle and hence we are getting an subtraction with overflow. In the other tests, with overwhelming probability, this will not happen.

**Comment by plafer** [comment]
From my understanding, the overflow is triggered when we attempt to read the same memory location in the same cycle - not terribly useful, but could happen due to user error (as mentioned in https://github.com/0xPolygonMiden/miden-vm/pull/1561#discussion_r1826016543)

**Comment by Al-Kindi-0** [comment]
Are you sure that the input stack is in the right order for `RCOMBBASE`?

**Comment by Al-Kindi-0** [comment]
> From my understanding, the overflow is triggered when we attempt to read the same memory location in the same cycle - not terribly useful, but could happen due to user error (as mentioned in [#1561 (comment)](https://github.com/0xPolygonMiden/miden-vm/pull/1561#discussion_r1826016543))

Yeah, maybe it is worth adding a disclaimer to the op to specify that the `Z_PTR` and `A_PTR` should be different

**Comment by Al-Kindi-0** [comment]
Yes, the stack input is not in the expected order i.e., the pointers are not in their correct place. This explains why Falcon tests didn't raise any errors as when the pointers are provided in their right place on the stack and the `Z_PTR` and `A_PTR` are different then everything should work well.
Thus the problem is only when by mistake the two aforementioned addresses are equal and this is what should be addressed here and in any future ops.

**Comment by plafer** [comment]
> Thus the problem is only when by mistake the two aforementioned addresses are equal and this is what should be addressed here and in any future ops.

Yup, exactly.

> Yeah, maybe it is worth adding a disclaimer to the op to specify that the Z_PTR and A_PTR should be different

I don't think so - they're obviously different from the documentation (i.e. one is randomness, the other is $T_i(...)$ etc). As mentioned a few times here, this issue here only addresses the case where there's a use error with `RCOMBBASE` (and potentially future ops).
